swift/apple/FirezoneKit/Sources/FirezoneKit/Managers/VPNProfileManager.swift

```
@@ -134,99 +129,96 @@ public class TunnelManager {
-      guard let managers = try? await NETunnelProviderManager.loadAllFromPreferences()
-      else {
-        Log.error(TunnelManagerError.cannotLoad)
-        return
-      }
+  func loadFromPreferences(vpnStateUpdateHandler: @escaping (NEVPNStatus, Settings?, String?) -> Void) async throws {
+    // loadAllFromPreferences() returns list of VPN profiles created by our main app's bundle ID.
+    // Since our bundle ID can change (by us), find the one that's current and ignore the others.
+    let managers = try await NETunnelProviderManager.loadAllFromPreferences()
 
-      Log.log("\(#function): \(managers.count) tunnel managers found")
-      for manager in managers {
-        if let protocolConfiguration = manager.protocolConfiguration as? NETunnelProviderProtocol,
-           protocolConfiguration.providerBundleIdentifier == TunnelManager.bundleIdentifier,
-           let providerConfiguration = protocolConfiguration.providerConfiguration as? [String: String]
-        {
-          // Found it
-          let settings = Settings.fromProviderConfiguration(providerConfiguration)
-          let actorName = providerConfiguration[TunnelManagerKeys.actorName]
-          if let internetResourceEnabled = providerConfiguration[TunnelManagerKeys.internetResourceEnabled]?.data(using: .utf8) {
-
-            self.internetResourceEnabled = (try? JSONDecoder().decode(Bool.self, from: internetResourceEnabled)) ?? false
+    Log.log("\(#function): \(managers.count) tunnel managers found")
+    for manager in managers {
+      if manager.localizedDescription == bundleDescription { // Found it
 
-          }
-          let status = manager.connection.status
+        guard let protocolConfiguration = manager.protocolConfiguration as? NETunnelProviderProtocol,
+              let providerConfiguration = protocolConfiguration.providerConfiguration as? [String: String]
+        else {
+          throw VPNProfileManagerError.cannotLoad
```

If we found a VPN Profile with our expected description (i.e. `Firezone`) but we couldn't load it, we now throw an explicit error instead of continuing to search.